django/db/models/sql/query.py

```diff
@@ -1992,7 +1982,12 @@ def split_exclude(self, filter_expr, can_reuse, names_with_path):
         lookup_class = select_field.get_lookup("exact")
         lookup = lookup_class(col, ResolvedOuterRef(trimmed_prefix))
         query.where.add(lookup, AND)
-        condition, needed_inner = self.build_filter(Exists(query))
+        condition, needed_inner = self.build_filter(
+            Exists(query),
+            current_negated=True,
+            branch_negated=True,
+            can_reuse=can_reuse,
```

Comment:
These [kwargs were originally passed along](https://github.com/django/django/commit/8593e162c9cb63a6c0b06daf045bc1c21eb4d7c1#diff-fd2300283d1546e36141373b0621f142ed871e3e8856e07efe5a22ecc38ad620L1813-R1811) but didn't carry any meaning since passing an expression to build_filter didn't make use of them until now.

Not doing so causes queries.tests.ManyToManyExcludeTest failures with the optimization enabled.